metaflow/tutorials/03-playlist-redux/playlist.py

```diff
@@ -76,18 +74,18 @@ def genre_movies(self):
 
         """
         from random import shuffle
+        from itertools import compress
 
         # For the genre of interest, generate a potential playlist using only
         # highest gross box office titles (i.e. those in the last quartile).
         genre = self.genre.lower()
         if genre not in self.genre_stats:
             self.movies = []
-
         else:
             df = self.genre_stats[genre]["dataframe"]
             quartiles = self.genre_stats[genre]["quartiles"]
-            selector = df["gross"] >= quartiles[-1]
-            self.movies = list(df[selector]["movie_title"])
+            selector = list(map(lambda x: x >= quartiles[-1], df["gross"]))
+            self.movies = list(compress(df["movie_title"], selector))
```

Comment:
aren't the two lines equivalent to this expression
```
[df['movie_title'][i] for i, g in enumerate(df['gross']) if g >= quartiles[-1]]
```
which is arguably simpler (no need for `list`, `map`, and `itertools.compress`)